compiler/rustc_middle/src/ty/error.rs

```diff
@@ -229,11 +229,17 @@ impl<'tcx> ty::TyS<'tcx> {
             ty::Foreign(def_id) => format!("extern type `{}`", tcx.def_path_str(def_id)).into(),
             ty::Array(t, n) => {
                 let n = tcx.lift(n).unwrap();
-                match n.try_eval_usize(tcx, ty::ParamEnv::empty()) {
-                    _ if t.is_simple_ty() => format!("array `{}`", self).into(),
-                    Some(n) => format!("array of {} element{}", n, pluralize!(n)).into(),
-                    None => "array".into(),
+                match t.is_simple_ty() {
```

Comment:
This would probably be better as an `if`.

Reply:
I personally prefer the match lol but I can change it if you really want it to be an if? ^^

Reply:
I think
```rust
if t.is_simple_ty() {
    return format!("array `{}`", self).into();
}
if let ty::ConstKind::Value(v) = n.val {
    if let Some(n) = v.try_to_machine_usize(tcx) {
        return format!("array of {} element{}", n, pluralize!(n)).into();
    }
}
"array".into()
```
is easier to read.